wiki/content/graphql/lambda/overview.md

```diff
@@ -0,0 +1,195 @@
+
+{{% notice "tip" %}}
+`self.addGraphQLResolvers` is the default choice for registering resolvers when not all the parent objects of a lambda field are required during resolution.
+{{% /notice %}}
+
+Each resolver function should return data in the exact format as the return type of GraphQL field, query, or mutation for which it is being registered.
+
+In the following example, the resolver function `myTypeResolver` registered for the `customField` field in `MyType` returns a string because the return type of that field in the GraphQL schema is `String`:
+
+```javascript
+const myTypeResolver = ({parent: {customField}}) => `My value is ${customField}.`
+
+self.addGraphQLResolvers({
+    "MyType.customField": myTypeResolver
+})
+```
+
+Another resolver example using a `graphql` call:
+
+```javascript
+async function todoTitles({ graphql }) {
+  const results = await graphql('{ queryTodo { title } }')
+  return results.data.queryTodo.map(t => t.title)
+}
+
+self.addGraphQLResolvers({
+  "Query.todoTitles": todoTitles
+})
+```
+
+### addMultiParentGraphQLResolvers
+
+The `self.addMultiParentGraphQLResolvers` is useful in scenarios where you want to perform computations involving all the parent objects returned from Dgraph for a lambda field.
+This is only useful for lambda fields and not for queries or mutations. 
+
+{{% notice "tip" %}}
+This method provides a much better performance if you are able to combine multiple requests together.
+{{% /notice %}}
+
+This method takes an object as an argument, which maps a resolver name to the resolver function that implements it. The resolver functions registered using this method receive `{ parents, args, graphql, dql }` as argument:
+
+- `parents`, a list of parent objects for which to resolve the current lambda field registered using `addMultiParentGraphQLResolvers`. Available only for types and interfaces (`null` for queries and mutations)
+- `args`,  the set of arguments for lambda queries and mutations (`null` for types/interfaces)
+- `graphql`, a function to execute auto-generated GraphQL API calls from the lambda server
+- `dql`, provides an API to execute DQL from the lambda server
+
+{{% notice "note" %}}
+This method should not be used for lambda queries or lambda mutations.
+{{% /notice %}}
+
+Each resolver function should return data as a list of the return type of GraphQL field for which it is being registered. 
+
+In the following example, the resolver function `rank()` registered for the `rank` field in `Author`, returns a list of integers because the return type of that field in the GraphQL schema is `Int`:
+
+```graphql
+type Author {
+    id: ID!
+    name: String! @search(by: [hash, trigram])
+    reputation: Float @search
+    rank: Int @lambda
+}
+```
+
+```javascript
+/* 
+This function computes the rank of each author based on the reputation of the author relative to other authors.
+*/
+async function rank({parents}) {
+    const idRepList = parents.map(function (parent) {
```

Comment:
can this be simplified (potentially by adding lodash? or something)

Reply:
the `rank()` function example was provided by @abhimanyusinghgaur. Do you mean simplifying the example code?

I'm not a JS expert so I'll ask for help to get a simplified example

Reply:
yes, exactly. I feel like you could do sortBy or something to simplify this code. It's a bit daunting.

Reply:
@abhimanyusinghgaur, could you help me out with this example `rank()` function, and make a simplified case for the doc?